runtime/__init__.py

```diff
@@ -0,0 +1,207 @@
+    )
+    print("Program methods:", program.method_names)
+    forward: Method = program.load_method("forward")
+
+    inputs = (torch.ones(2, 2), torch.ones(2, 2))
+    outputs = forward.execute(inputs)
+    print(f"Ran forward({inputs})")
+    print(f"  outputs: {outputs}")
+
+Example output:
+.. code-block:: text
+
+    Program methods: ('forward', 'forward2')
+    Ran forward((tensor([[1., 1.],
+            [1., 1.]]), tensor([[1., 1.],
+            [1., 1.]])))
+      outputs: [tensor([[1., 1.],
+            [1., 1.]])]
+"""
+
+import functools
+from pathlib import Path
+from types import ModuleType
+from typing import Any, BinaryIO, Sequence, Union
+
+try:
+    from executorch.extension.pybindings.portable_lib import (
+        ExecuTorchModule,
+        MethodMeta,
+        Verification,
+    )
+except ModuleNotFoundError as e:
+    raise ModuleNotFoundError(
+        "Prebuilt <site-packages>/extension/pybindings/_portable_lib.so "
+        "is not found. Please reinstall ExecuTorch from pip."
+    ) from e
+
+
+class Method:
+    """An ExecuTorch method, loaded from a Program.
+    This can be used to execute the method with inputs.
+    """
+
+    def __init__(self, method_name: str, module: ExecuTorchModule) -> None:
+        # TODO: This class should be pybind to the C++ counterpart instead of hosting ExecuTorchModule.
+        self._method_name = method_name
+        self._module = module
+
+    def execute(self, inputs: Sequence[Any]) -> Sequence[Any]:
+        """Executes the method with the given inputs.
+
+        Args:
+            inputs: The inputs to the method.
+
+        Returns:
+            The outputs of the method.
+        """
+        return self._module.run_method(self._method_name, inputs)
+
+    @property
+    def metadata(self) -> MethodMeta:
+        """Gets the metadata for the method.
+
+        Returns:
+            The metadata for the method.
+        """
+        return self._module.method_meta(self._method_name)
+
+
+class Program:
+    """An ExecuTorch program, loaded from binary PTE data.
+
+    This can be used to load the methods/models defined by the program.
+    """
+
+    def __init__(self, module: ExecuTorchModule, data: bytes) -> None:
+        # Hold the data so the program is not freed.
+        self._data = data
+        self._module = module
+        self._methods = {}
+        # ExecuTorchModule already pre-loads all Methods when created, so this
+        # doesn't do any extra work. TODO: Don't load a given Method until
+        # load_method() is called. Create a separate Method instance each time,
+        # to allow multiple independent instances of the same model.
+        for method_name in self._module.method_names():
+            self._methods[method_name] = Method(method_name, self._module)
+
+    @property
+    def method_names(self) -> Sequence[str]:
+        return set(self._methods.keys())
+
+    def load_method(self, name: str) -> Method:
```

Comment:
Since this can return None. The "Returns" section should also mention the None case.
```suggestion
    def load_method(self, name: str) -> Optional[Method]:
```